src/Core/Collections/Leaderboard/Leaderboard.tsx

```diff
@@ -56,14 +65,27 @@ const Leaderboard = (): ReactElement => {
     leaderboardTimeRange === LeaderboardTimeRange.IGBO_VOICE_ATHON &&
     !moment().isBetween(igboVoiceathonStartDate, igboVoiceathonEndDate);
 
-  const handleUpdateLeaderboard = (event) => {
-    const updatedLeaderboard = event.target.value as LeaderboardType;
-    if (leaderboard !== updatedLeaderboard) {
-      setLeaderboard(updatedLeaderboard);
+  // const handleUpdateLeaderboard = (event) => {
+  //   const updatedLeaderboard = event.target.value as LeaderboardType;
+  //   if (leaderboard !== updatedLeaderboard) {
+  //     setLeaderboard(updatedLeaderboard);
+  //   }
+  // };
+  const handleLeaderboardClick = (index: number) => () => {
+    const leaderboardItem = leaderboardItems[index];
+    if (leaderboardItem.uid !== leaderboardItems[leaderboardIndex]?.uid) {
+      setLeaderboardIndex(index);
```

Comment:
you don't need to check to see if the uid is selected already, you can just call setSelectedLeaderboard

```suggestion
    setSelectedLeaderboard(leaderboardId);
```